### PR TITLE
Added a selection property for a special communication pattern

### LINKED ISSUE #441: Questions on Connection Groups
A couple of questions we might want to address regarding Connection Groups.

1) Can the application query the interface to figure if two connection in a group share the same transport connection but on different streams or not. Is that useful? I thinks we should explicitly say that if a multiplex protocol is used, calling Clone() MUST create a new stream (and not a new connection).

2) If a new connection is created by a Listener it is automatically part of a group, right? Can I detach that connection from the group? Otherwise how can I change a property only for that connection? Isn't is actually confusing that these connection are grouped?

2) Should we add a pointer to coupled congestion control (rmcat) in that section?

### PR DESCRIPTION
**Pull request by mwelzl**
.. that gets in the way of using streams when cloning.
Closes #441.

### CODE REVIEW

File: draft-ietf-taps-interface.md [hunk 1]
```diff
@@ -1124,6 +1125,30 @@ not all ICMP errors will necessarily be delivered, so applications cannot rely
...
+
+Default:
+: Ignore
+
+The most common client-server communication pattern involves the
+client actively opening a connection, then sending data to the server. The
+server listens (passive open), reads, and then answers. This property
+specifies whether an application wants to diverge from this pattern -- either
+by actively opening, immediately followed by reading, or passively opening,
+immediately followed by writing.
```
Comment (philsbln) [inline draft-ietf-taps-interface.md:1144]: I get the problem for read-after-active-open, but what is the problem with passive open and immediately writing – this is fairly common in TCP based protocols and you can easily put the write in the connection ready.

Comment (mwelzl) [reply draft-ietf-taps-interface.md:1144]: It's the same thing with multi-streaming: a stream only comes into being when the first message is sent on it. So, "passive open and immediately writing" doesn't map onto anything with streams - there is no such thing as a stream becoming available without data arriving on it.

File: draft-ietf-taps-interface.md [hunk 2]
```diff
@@ -1124,6 +1125,30 @@ not all ICMP errors will necessarily be delivered, so applications cannot rely
 on receiving them.
 
 
+### Active side is not the first to write {#active-read-before-send}
```
Comment (philsbln) [inline draft-ietf-taps-interface.md:1128]:
```suggestion
### Initiating side is not the first to write {#active-read-before-send}
```

**Comment by csperkins** [comment]
How does this interact with `Rendezvous()`?

File: draft-ietf-taps-interface.md [hunk 3]
```diff
@@ -1124,6 +1125,30 @@ not all ICMP errors will necessarily be delivered, so applications cannot rely
...
+
+Name:
+: activeReadBeforeSend
+
+Type:
+: Preference
+
+Default:
+: Ignore
+
```
Comment (csperkins) [inline draft-ietf-taps-interface.md:1138]: I wonder if this should be `Type: Enumeration`, with values `InitiatorSendsFirst` (the default) and `ListenerSendsFirst`?

Comment (mwelzl) [reply draft-ietf-taps-interface.md:1138]: Why make it so limiting? Preference allows for values such as Prefer, Avoid, Require, ...

**Comment by mwelzl** [comment]
@csperkins - this now states that it is irrelevant for Rendezvous(), and explicitly mentions the Initiate() and Listen() calls when talking about actively and passively opening a Connection.

Why is it irrelevant?

Remember, the problem is that you actively open a Connection that could be mapped to a stream, in which case an active Open doesn't do anything on the wire - and then, right away, you want to read. Then the peer never heard from you, you haven't even tried sending anything to it.

Enabling this property will allow the TAPS system to carry out such a Connection = stream mapping. When calling rendezvous(), however, just using a stream is out of the question - you need signaling, and chances are that you're going to actively send stuff in any case. In other words, I don't think there's a mapping that would work with rendezvous() that enabling this property would allow, and hence the property is irrelevant for rendezvous().

**Comment by mwelzl** [comment]
Merging this now: I think I have addressed the comment by @csperkins. We can always fix things later again if we must - but for now, this closes #441, and it also relates to #508, which gates #507.... so it's probably better to move forward at this point.

File: draft-ietf-taps-interface.md (hunk 3 above)
Comment (csperkins) [reply draft-ietf-taps-interface.md:1138]: How does a preference work here? If the application protocol requires the listener to send first, then it need to do this or fail.

File: draft-ietf-taps-interface.md [hunk 4]
```diff
@@ -1124,6 +1125,31 @@ not all ICMP errors will necessarily be delivered, so applications cannot rely
...
+The most common client-server communication pattern involves the
+client actively opening a connection, then sending data to the server. The
+server listens (passive open), reads, and then answers. This property
+specifies whether an application wants to diverge from this pattern -- either
+by actively opening with Initiate(), immediately followed by reading, or passively opening with Listen(),
+immediately followed by writing. This property is irrelevant for connection
+establishment using Rendezvous().
```
Comment (csperkins) [inline draft-ietf-taps-interface.md:1145]: "irrelevant" isn't right here - maybe "is ignored"?

Comment (mwelzl) [reply draft-ietf-taps-interface.md:1145]: ACK, fixed with a commit to the master

**Pull request merged**